### PR TITLE
Provide better telemetry out-of-the-box

### PR DESCRIPTION
**Pull request by tomkerkhove**
Signed-off-by: Tom Kerkhove <kerkhove.tom@gmail.com>

### CODE REVIEW

File: src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs [hunk 1]
```diff
@@ -104,7 +104,7 @@ public static IEnumerable<MessageHandler> SubtractFrom(IServiceProvider serviceP
 
             if (actualMessageContextType == expectedMessageContextType)
             {
-                _logger.LogInformation(
+                _logger.LogTrace(
                     "Message context type '{ActualMessageContextType}' matches registered message handler's {MessageHandlerType} context type {ExpectedMessageContextType}",
                     actualMessageContextType.Name, ServiceType.Name, expectedMessageContextType.Name);
```
Comment (stijnmoreels) [inline src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs:109]:
```suggestion
#if DEBUG
                _logger.LogTrace(
                    "Message context type '{ActualMessageContextType}' matches registered message handler's {MessageHandlerType} context type {ExpectedMessageContextType}",
                    actualMessageContextType.Name, ServiceType.Name, expectedMessageContextType.Name);
#endif
```

File: src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs [hunk 2]
```diff
@@ -119,7 +119,7 @@ public static IEnumerable<MessageHandler> SubtractFrom(IServiceProvider serviceP
                         BindingFlags.Instance | BindingFlags.NonPublic,
                         messageContext);
 
-                    _logger.LogInformation(
+                    _logger.LogTrace(
                         "Message context predicate registered with the message handler {MessageHandlerType} resulted in {Result}, so {Action} process this message",
                         ServiceType.Name, canProcessMessage, canProcessMessage ? "can" : "can't");
```
Comment (stijnmoreels) [inline src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs:124]:
```suggestion
#if DEBUG
                    _logger.LogTrace(
                        "Message context predicate registered with the message handler {MessageHandlerType} resulted in {Result}, so {Action} process this message",
                        ServiceType.Name, canProcessMessage, canProcessMessage ? "can" : "can't");
#endif                        
```

Same as below.

File: src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs [hunk 3]
```diff
@@ -130,7 +130,7 @@ public static IEnumerable<MessageHandler> SubtractFrom(IServiceProvider serviceP
                 return true;
             }
 
-            _logger.LogInformation(
+            _logger.LogTrace(
                 "Message context type '{ActualMessageContextType}' doesn't matches registered message handler's {MessageHandlerType} context type {ExpectedMessageContextType}",
                 actualMessageContextType.Name, ServiceType.Name, expectedMessageContextType.Name);
```
Comment (stijnmoreels) [inline src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs:135]:
```suggestion
#if DEBUG
            _logger.LogTrace(
                "Message context type '{ActualMessageContextType}' doesn't matches registered message handler's {MessageHandlerType} context type {ExpectedMessageContextType}",
                actualMessageContextType.Name, ServiceType.Name, expectedMessageContextType.Name);
#endif                
```

Same as below.

**Review by stijnmoreels — APPROVED** [review]

File: src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs (hunk 1 above)
Comment (tomkerkhove) [reply src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs]: No I think it's ok, but only as a trace

File: src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs (hunk 2 above)
Comment (tomkerkhove) [reply src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs]: No I think it's ok, but only as a trace

File: src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs (hunk 3 above)
Comment (tomkerkhove) [reply src/Arcus.Messaging.Pumps.Abstractions/MessageHandling/MessageHandler.cs]: No I think it's ok, but only as a trace

**Pull request merged**